tests/helpers.ts

```diff
@@ -1,55 +1,60 @@
+/**
+ * @license
+ * Copyright 2019 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 import * as nock from 'nock';
 import * as path from 'path';
 import * as balenaCLI from '../build/app';
 
 export const runCommand = async (cmd: string) => {
 	const preArgs = [process.argv[0], path.join(process.cwd(), 'bin', 'balena')];
 
-	const oldStdOut = process.stdout.write;
-	const oldStdErr = process.stderr.write;
-
 	const err: string[] = [];
 	const out: string[] = [];
 
-	// @ts-ignore
-	process.stdout.write = (log: string) => {
+	const stdoutHook = (log: string | Buffer) => {
 		// Skip over debug messages
-		if (!log.startsWith('[debug]')) {
+		if (typeof log === 'string' && !log.startsWith('[debug]')) {
 			out.push(log);
 		}
-		oldStdOut(log);
 	};
-	// @ts-ignore
-	process.stderr.write = (log: string) => {
+	const stderrHook = (log: string | Buffer) => {
 		// Skip over debug messages
 		if (
+			typeof log === 'string' &&
 			!log.startsWith('[debug]') &&
 			// TODO stop this warning message from appearing when running
 			// sdk.setSharedOptions multiple times in the same process
 			!log.startsWith('Shared SDK options')
 		) {
 			err.push(log);
 		}
-		oldStdErr(log);
 	};
+	const unhookIntercept = require('intercept-stdout')(stdoutHook, stderrHook);
```

Comment:
Can you import this dependency at the top of the file using `import` instead of an inline `require`?

Reply:
Check my change: is it good enough to have a `require()` at the top of the function? Placing an import at the top of the file:
```
import * as intercept from 'intercept-stdout';
```
Results in:
```
TSError: ⨯ Unable to compile TypeScript:
tests/helpers.ts:18:28 - error TS7016: Could not find a declaration file for module 'intercept-stdout'. 'balena-io/balena-cli/node_modules/intercept-stdout/intercept-stdout.js' implicitly has an 'any' type.
```

Reply:
@pdcastro We should do one of the following:
1. Stub the typings for the `intercept-stdout` module in the CLI
2. PR typings to the `intercept-stdout` repo
3. PR typings for the `intercept-stdout` module to DefinitelyTyped

We could in fact do all three of these things, start with 1), if 2) fails, do 3).

Reply:
As requested: added file `typings/intercept-stdout/index.d.ts` to `balena-cli`, and also created a PR at `sfarthin/intercept-stdout`: https://github.com/sfarthin/intercept-stdout/pull/14

Reply:
As there wasn't activity on [the PR for the intercept-stdout package](https://github.com/sfarthin/intercept-stdout/pull/14) after a week, I've progressed to step 3 and created a PR against DefinitelyTyped: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38373